app/src/main/java/io/intrepid/contest/screens/join/JoinPresenter.java

```diff
@@ -41,17 +43,32 @@ public void onSubmitButtonClicked(String code) {
 
                     // TODO: once API endpoing works, stop showing message and skipping to next screen
                     view.showMessage(R.string.error_api);
-                    view.showEntryNameScreen();
+                    InvitationResponse response = new InvitationResponse();
+                    response.participant = new Participant();
+                    response.participant.setContestId(UUID.randomUUID());
+                    if (code.equals("judge")) {
+                        response.participant.setParticipationType(ParticipationType.JUDGE);
+                    } else {
+                        response.participant.setParticipationType(ParticipationType.CONTESTANT);
+                    }
+                    showResult(response);
                 });
         disposables.add(disposable);
     }
 
     private void showResult(InvitationResponse response) {
```

Comment:
can we call this method onResponse or onInvitationResponseReceived etc. I think it does more than just show the result.